src/store/modules/wallet.js

```diff
@@ -413,6 +425,119 @@ const actions = {
+    const getFolloweeEvents = followee =>
+      this.$api
+        .$get(
+          getNFTEvents({
+            involver: followee,
+            limit: WALLET_EVENT_LIMIT,
+            actionType: ['/cosmos.nft.v1beta1.MsgSend', 'buy_nft', 'new_class'],
+            ignoreToList: LIKECOIN_NFT_API_WALLET,
+            reverse: true,
+          })
+        )
+        .then(res => res.events)
+        .catch(() => []);
+    const eventPromises = followees.map(getFolloweeEvents);
+    const responses = await Promise.all(eventPromises);
+    const events = responses.flat();
+    events.sort((a, b) => new Date(b.timestamp) - new Date(a.timestamp));
+    const categorizeEvent = event => {
+      switch (event.action) {
+        case '/cosmos.nft.v1beta1.MsgSend':
+          event.type =
+            event.sender === LIKECOIN_NFT_API_WALLET ? 'collect' : 'send';
+          break;
+        case 'new_class':
+          event.type = 'publish';
+          break;
+        case 'buy_nft':
+          event.type = 'purchase';
+          break;
+        default:
+          break;
+      }
+      return event;
+    };
+    const categorizedEvents = events.map(categorizeEvent);
+
+    // get creator message for 'publish' events
+    const publishEvents = categorizedEvents.filter(
+      event => event.type === 'publish'
+    );
+    await Promise.all(
+      publishEvents.map(async event => {
+        const metadata = await dispatch(
+          'lazyGetNFTClassMetadata',
+          event.class_id
+        );
+        if (metadata.message) {
+          event.memo = metadata.message;
+        }
+      })
+    );
+    publishEvents.forEach(publishEvent => {
+      const index = categorizedEvents.findIndex(
+        event => event === publishEvent
+      );
+      if (index !== -1) {
+        categorizedEvents[index] = publishEvent;
+      }
+    });
+
+    // get collector message for 'collect' events
+    const collectEvents = categorizedEvents.filter(
+      event => event.type === 'collect'
+    );
+    const promises = collectEvents.map(e =>
+      getNFTHistoryDataMap({
+        axios: this.$api,
+        classId: e.class_id,
+        txHash: e.tx_hash,
+      })
+    );
+    const historyDatas = await Promise.all(promises);
+    historyDatas.forEach((m, index) => {
+      m.forEach(data => {
+        const { granterMemo, price, sellerWallet } = data;
+        collectEvents[index].price = price;
+        collectEvents[index].granterMemo = granterMemo;
+        collectEvents[index].sender = sellerWallet;
+      });
+    });
+    categorizedEvents.forEach(event => {
+      if (event.type === 'collect') {
+        Object.assign(
+          event,
+          collectEvents.find(collectEvent => collectEvent === event)
+        );
+      }
+    });
+
+    // filter available events
+    const filteredEvents = categorizedEvents.filter(event => {
+      if (event.sender === address || event.receiver === address) {
+        return false;
```

Comment:
Why filter receiver is self events?

Reply:
We can currently see events where the recipient is yourself from the [notifications /]